### PR TITLE
rebuilt AVS verification code

### PR DESCRIPTION
**Pull request by fiestacasey**
@ksolkowski @kahlya @mikeSheets 

please go through each of my new constants and compare with the document here https://drive.google.com/file/d/0B3jCb2M9MdrKRVFXT043LUtuaDg/view

Ensure that everything is in the right place and that I didn’t leave anything out. 

R is intentionally left out because we want it to fail with “unknown” error and the user to try again

### CODE REVIEW

File: lib/card_connect_gateway/authorization/response.rb [hunk 1]
```diff
@@ -39,6 +39,177 @@ class Response < BaseResponse
...
+
+      VISA_ADDRESS_MISMATCH_CODES = [
+        C, # Street address and postal code not verified due to incompatible formats.
+        N, # No match. Acquirer sent postal/ZIP code only, or, street address only, or, both postal code and street address. Also used when acquirer requests AVS, but sends no AVS data in Field 123.
+        Z  # Postal/ZIP matches; street address does not match or Street address not included in request.
+      ]
+
+      VISA_ZIPCODE_MISMATCH_CODES = [
+        A, # Address matches, ZIP does not.
+        B, # Postal code not verified due to incompatible formats.
```
Comment (kahlya) [inline lib/card_connect_gateway/authorization/response.rb]: Document says this is 
```Street addresses match. Postal code not verified due to incompatible
formats.``` 
Don't know if the street addresses match part is important though.

File: lib/card_connect_gateway/authorization/response.rb [hunk 2]
```diff
@@ -39,6 +39,177 @@ class Response < BaseResponse
...
+        C, # Street address and postal code not verified due to incompatible formats.
+        N  # No match.
+      ]
+
+      VISA_ADDRESS_UNKNOWN_CODES = [
+        G, # Address information not verified for international transaction
+        I, # Address information not verified,
+        P, # street address not verified due to incompatible formats.
```
Comment (kahlya) [inline lib/card_connect_gateway/authorization/response.rb]: Like above, this one includes ``` Postal Code Match ```

**Comment by ksolkowski** [comment]
fancy! 👍

**Comment by kahlya** [comment]
looks 👍

**Pull request merged**